### PR TITLE
fix: resolve broken event_bus import in token refresh

### LINKED ISSUE #1542: Bug: Missing event_bus module causes agent registration failure (SDR stays Pending)
### Summary
`application_sdk/execution/_temporal/auth.py` imports from a non-existent module `application_sdk.execution._temporal.interceptors.event_bus`, causing agent heartbeat/registration events to silently fail. This results in SDR deployments showing **Pending** status in Atlan UI instead of **Active**, even though the pod is healthy.

### Error
```
ModuleNotFoundError: No module named 'application_sdk.execution._temporal.interceptors.event_bus'
```

### Full Error Trace
```
File "/app/.venv/lib/python3.13/site-packages/application_sdk/execution/_temporal/auth.py", line 233, in _emit_token_refresh_event
    from application_sdk.execution._temporal.interceptors.event_bus import (
ModuleNotFoundError: No module named 'application_sdk.execution._temporal.interceptors.event_bus'
```

This fires every 5 minutes on token refresh, logged as:
```
WARNING - Failed to emit token_refresh event
```

### Root Cause
In [`auth.py` (lines 233-234)](https://github.com/atlanhq/application-sdk/blob/main/application_sdk/execution/_temporal/auth.py#L233), `_emit_token_refresh_event()` imports:
```python
from application_sdk.execution._temporal.interceptors.event_bus import (
    get_event_bus,
)
from application_sdk.execution._temporal.interceptors.events import (
    TOKEN_REFRESH,
    LifecycleEvent,
)
```

However:
1. **`event_bus.py` does not exist** — the `interceptors/` directory only contains: `__init__.py`, `activity_failure_logging.py`, `correlation_interceptor.py`, `events.py`, `execution_context_interceptor.py`, `lock.py`, `outputs.py`
2. **`TOKEN_REFRESH` and `LifecycleEvent`** are not defined in `events.py` either
3. **`get_event_bus`** is not defined anywhere in the codebase (confirmed via GitHub code search)

These symbols were referenced but never implemented.

### How We Found It
1. Deployed `alloydb-postgres-app` (image: `revert-workarounds-84c0d99`) on SDR
2. Pod was `1/1 Running`, health check `200 OK`, 0 restarts
3. Atlan UI showed agent status as **Pending** instead of **Active**
4. Checked logs and found `ModuleNotFoundError` firing every 5 minutes on token refresh
5. Verified the same error exists in `cloudsql-postgres-agent-devex` deployment (image: `dbbi-331-6babdb6`) — also stuck on Inactive
6. Confirmed `event_bus.py` was never created on `main` branch

### Impact
- **All SDR deployments using SDK >= 3.x** are affected
- Agent status stays Pending/Inactive in Atlan UI
- Heartbeat/registration events are never published
- The error is caught silently (WARNING log), so the app runs fine otherwise

### Expected Fix
Either:
1. Create the missing `event_bus.py` module with `get_event_bus()`, and add `TOKEN_REFRESH`/`LifecycleEvent` to `events.py`
2. Or refactor `auth.py` to use the existing event publishing mechanism in `events.py`

### Environment
- **Tenant:** apps-typedef.atlan.com, devex.atlan.com
- **Helm chart:** alloydb-postgres-app v0.1.11, cloudsql-postgres-app
- **SDK:** `fix/dapr-binding-double-json-encoding` branch (same code is on `main`)
- **Python:** 3.13
- **Dapr:** v1.17.5

### PR DESCRIPTION
**Pull request by SwarnadeepChatterjee**
## Summary

- Fixes `_emit_token_refresh_event` in `auth.py` which imported from a non-existent module `application_sdk.execution._temporal.interceptors.event_bus` and referenced undefined symbols (`get_event_bus`, `TOKEN_REFRESH`, `LifecycleEvent`)
- Refactored to use the existing `_publish_event_via_binding` function from `interceptors/events.py` and standard `Event`/`ApplicationEventNames` types from `contracts/events.py`, matching the pattern used by all other event publishers in the codebase
- `ApplicationEventNames.TOKEN_REFRESH` already existed in `contracts/events.py`, so no new event types were needed
- Added 7 unit tests covering: successful event publishing, graceful failure handling (runtime and import errors), correct event type/name values, expected metadata fields, deployment_name fallback, and None expires_at edge case

Closes #1542

## Test plan

- [x] Unit tests for `_emit_token_refresh_event` added in `tests/unit/execution/test_auth_token_refresh_event.py` (7 tests, all passing)
- [ ] Verify `_emit_token_refresh_event` no longer raises `ModuleNotFoundError` at runtime
- [ ] Confirm token refresh events are published via the Dapr event binding when auth is configured
- [ ] Verify the event payload contains the expected fields (`force_refresh`, `token_expiry_time`, `time_until_expiry`, `refresh_timestamp`, `application_name`, `deployment_name`)
- [x] Pre-commit checks (ruff, pyright, isort) pass

### CODE REVIEW

**Review by Aryamanz29 — DISMISSED** [review]
LGTM - could you please also add test for this change? 🙏

**Comment by vaibhavatlan** [comment]
@sdk-review

**Review by Aryamanz29 — APPROVED** [review]
LGTM - thanks! @SwarnadeepChatterjee

File: tests/unit/execution/test_auth_token_refresh_event.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            "ATLAN_DEPLOYMENT_NAME": "my-deployment",
+        }
+
+        with (
+            mock.patch.dict("os.environ", env_vars, clear=False),
+            mock.patch(_PUBLISH_TARGET) as mock_publish,
+        ):
+            mock_publish.return_value = None
+
+            await manager._emit_token_refresh_event(expires_at)
```
[the remark this answers is not in the document]
Comment (vaibhavatlan) [reply tests/unit/execution/test_auth_token_refresh_event.py:97]: Ignore this pls

**Review by vaibhavatlan — APPROVED** [review]
LGTM
